File: fabric/src/main/java/juuxel/adorn/platform/fabric/TagBridgeImpl.java
package juuxel.adorn.platform.fabric;

import net.fabricmc.fabric.api.tag.TagRegistry;
import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraft.tag.Tag;
import net.minecraft.util.Identifier;
import org.jetbrains.annotations.NotNull;

public final class TagBridgeImpl {
    @NotNull
    public static Tag<Block> block(Identifier id) {
        return TagRegistry.block(id);
    }

    @NotNull
    public static Tag<Item> item(Identifier id) {
        return TagRegistry.item(id);
    }
}

File: fabric/src/main/kotlin/juuxel/adorn/platform/fabric/TagBridgeImpl.kt
package juuxel.adorn.platform.fabric

import juuxel.adorn.platform.TagBridge
import net.fabricmc.fabric.api.tag.TagFactory
import net.minecraft.block.Block
import net.minecraft.item.Item
import net.minecraft.tag.Tag
import net.minecraft.util.Identifier

object TagBridgeImpl : TagBridge {
    override fun block(id: Identifier): Tag<Block> =
        TagFactory.BLOCK.create(id)

    override fun item(id: Identifier): Tag<Item> =
        TagFactory.ITEM.create(id)
}
